src/GovUk.Education.ExploreEducationStatistics.Common/Services/FileStorageUtils.cs

```diff
@@ -92,15 +92,18 @@ public static CloudBlockBlob GetBlockBlob(string storageConnectionString, string
         }
 
         public static IEnumerable<FileInfo> ListPublicFilesPreview(string storageConnectionString, string containerName,
-            Guid releaseId)
+            IEnumerable<Guid> referencedVersions)
         {
             var files = new List<FileInfo>();
 
-            files.AddRange(ListFiles(storageConnectionString, containerName,
-                AdminReleaseDirectoryPath(releaseId, ReleaseFileTypes.Data), false));
+            foreach (var version in referencedVersions)
```

Comment:
We need to get the release concerned because that's the dir we need to refernce in storage & the file reference references back to the original release id that it was uploaded against but ReleaseFiles contains the link that connect them to the current amendment.